### PR TITLE
Fix file extension replacement for resource paths when using static frameworks

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/10206

### CODE REVIEW

File: lib/cocoapods/target/aggregate_target.rb [hunk 1]
```diff
@@ -292,7 +292,7 @@ def resource_paths_by_config
                 extname = File.extname(resource_path)
                 if self.class.resource_extension_compilable?(extname)
                   output_extname = self.class.output_extension_for_resource(extname)
-                  built_product_dir.join(File.basename(resource_path, extname)).sub_ext(output_extname).to_s
```
Comment (igor-makarov) [inline lib/cocoapods/target/aggregate_target.rb:295]: @dnkoutso 
This is actually unnecessary. I realize I've used `sub_ext` incorrectly in my impl.  
The correct use would be: 
```
built_product_dir.join(resource_path).sub_ext(output_extname).to_s
```

File: lib/cocoapods/target/aggregate_target.rb [hunk 2]
```diff
@@ -292,7 +292,7 @@ def resource_paths_by_config
...
+                  built_product_dir.join(change_ext(File.basename(resource_path), output_extname)).to_s
```
Comment (igor-makarov) [inline lib/cocoapods/target/aggregate_target.rb:295]: @dnkoutso 
This is actually unnecessary. I realize I've used `sub_ext` incorrectly in my impl.  
The correct use would be: 
```
built_product_dir.join(resource_path).sub_ext(output_extname).to_s
```

Comment (dnkoutso) [reply lib/cocoapods/target/aggregate_target.rb:295]: updating!

Comment (dnkoutso) [reply lib/cocoapods/target/aggregate_target.rb]: @igor-makarov fixed! the test passes too with your recommendation.

Comment (dnkoutso) [reply lib/cocoapods/target/aggregate_target.rb]: actually @igor-makarov this is not correct still:

Screen_Shot_2020-12-31_at_11_28_34_AM

Comment (dnkoutso) [reply lib/cocoapods/target/aggregate_target.rb]: I think we need:

```ruby
                  built_product_dir.join(File.basename(resource_path)).sub_ext(output_extname).to_s
```

Comment (dnkoutso) [reply lib/cocoapods/target/aggregate_target.rb]: also updated tests to catch this.

Comment (igor-makarov) [reply lib/cocoapods/target/aggregate_target.rb]: Yes! This is the right one!

**Review by igor-makarov — APPROVED** [review]

**Pull request merged**